### PR TITLE
chore: use jest-circus

### PR DESCRIPTION
**Pull request by viceice**
* use modern `jest-circus` instead `jest-jasmine2` runner
* use `node` jest environment instead of `jsdom` (we don't need a browser dom)

This reduces my local test time from ~70s to ~60s for full test.

These settings will become default in jest [27](https://jestjs.io/blog/2020/05/05/jest-26)

### CODE REVIEW

**Comment by JamieMagee** [comment]
What happened to the perf improvements? 😂

**Review by JamieMagee — DISMISSED** [review]

**Comment by viceice** [comment]
strange, timeout error. retrying

**Comment by viceice** [comment]
@JamieMagee Any idea why the new timeout doen't work?

**Review by JamieMagee — DISMISSED** [review]

**Comment by viceice** [comment]
got the timeout working 💪

**Comment by rarkins** [comment]
Needs de-conflicting :)

**Comment by viceice** [comment]
done 😅

File: lib/manager/gradle-wrapper/artifacts.spec.ts [hunk 1]
```diff
@@ -19,8 +19,6 @@ const config = {
   toVersion: '5.6.4',
 };
 
-jest.mock('../../platform');
```
Comment (rarkins) [inline lib/manager/gradle-wrapper/artifacts.spec.ts:22]: Is this related?

Comment (viceice) [reply lib/manager/gradle-wrapper/artifacts.spec.ts:22]: no, simply not nessessary, because platform is always mocked by default in `test/globals.ts`

Comment (viceice) [reply lib/manager/gradle-wrapper/artifacts.spec.ts]: removed, will be removed in #6233 anyways
